### PR TITLE
Feature/cmake make program

### PR DESCRIPTION
**Pull request by lasote**
Merge with  

https://github.com/conan-io/conan/issues/2249
https://github.com/conan-io/docs/pull/514
https://github.com/conan-io/conan/pull/2431

### CODE REVIEW

File: conan/external_tools/bare_mingw_test.py [hunk 1]
```diff
@@ -55,3 +56,21 @@ def build_mingw_test(self):
...
+class MinGWUsingCMakeAndMake(unittest.TestCase):
+
+    def build_hello_with_make_test(self):
+        if Version(client_version) < Version("1.1.0-dev"):
+            raise nose.SkipTest('Only >= 1.1.0-dev')
+
+        client = TestClient()
+        files = cpp_hello_conan_files("lib", "1.0")
+        client.save(files)
+        make_location = "/usr/bin/make" if not tools.os_info.is_windows else CONAN_STANDALONE_MAKE_PATH
```
Comment (memsharded) [inline conan/external_tools/bare_mingw_test.py]: Shouldn't be ``CONAN_STANDALONE_MAKE_PATH`` the one that is conditionally set in conf to either "/usr/bin/make" or to "C:/mingw-w64/mingw32/bin/mingw32-make.exe" depending on the system?

Comment (lasote) [reply conan/external_tools/bare_mingw_test.py]: Sure

**Comment by danimtb** [comment]
Should this be merged now that all the above tasks were completed?

**Comment by danimtb** [comment]
Closed as obsolete

**Pull request closed without merging**